internal/engine/local_container_build_and_deployer.go

```diff
@@ -30,22 +30,26 @@ func NewLocalContainerBuildAndDeployer(cu *build.ContainerUpdater,
 }
 
 func (cbd *LocalContainerBuildAndDeployer) BuildAndDeploy(ctx context.Context, st store.RStore, specs []model.TargetSpec, stateSet store.BuildStateSet) (store.BuildResultSet, error) {
-	iTargets, kTargets := extractImageAndK8sTargets(specs)
-	if len(kTargets) != 1 || len(iTargets) != 1 {
-		return store.BuildResultSet{}, RedirectToNextBuilderf(
-			"LocalContainerBuildAndDeployer requires exactly one image spec and one k8s deploy spec")
+	iTargets, err := extractImageTargetsForLiveUpdates(specs, stateSet)
```

Comment:
in theory, this build and deployer should work fine with docker-compose, but i haven't verified it all yet manually.